src/Microsoft.AspNetCore.Authentication.MicrosoftAccount/MicrosoftAccountOptions.cs

```diff
@@ -30,9 +30,7 @@ public MicrosoftAccountOptions()
             ClaimResolvers.Add(ClaimTypes.Name, "displayName");
             ClaimResolvers.Add(ClaimTypes.GivenName, "givenName");
             ClaimResolvers.Add(ClaimTypes.Surname, "surname");
-            // Last one wins, if present
-            ClaimResolvers.Add(ClaimTypes.Email, "userPrincipalName");
-            ClaimResolvers.Add(ClaimTypes.Email, "mail");
+            ClaimResolvers.AddCustom(ClaimTypes.Email, user => user.Value<string>("mail") ?? user.Value<string>("userPrincipalName"));
```

Comment:
Honestly I almost prefer if we didn't have the sugar and only exposed the lambda syntax to be clear, its super obvious what's going on that way and we can expose helpers as needed to make the nested stuff nicer